test/hotel_system_test.rb

```diff
@@ -0,0 +1,358 @@
+      expect {
+        @hotel_system.create_hotel_block(rooms: rooms4, date_range: date_range4, discount: @discount_rate)
+      }.must_raise ArgumentError
+    end
+    
+    it "raises ArgumentError when user creates a block that has a booked room for the given date range" do
+      rooms3 = [2, 3, 4]
+      start_date3 = @end_date
+      end_date3 = start_date3 + @duration
+      
+      date_range3 = Hotel::DateRange.new(start_date3, end_date3)
+      @hotel_system.create_hotel_block(rooms: rooms3, date_range: date_range3, discount: @discount_rate)
+      expect (@hotel_system.hotel_blocks).wont_be_empty
+      
+      # make reservation on the first room available in this date range (room #1)
+      @hotel_system.make_reservation(start_date3, end_date3)
+      
+      rooms4 = [1, 12, 13]
+      expect {
+        @hotel_system.create_hotel_block(rooms: rooms4, date_range: date_range3, discount: @discount_rate)
+      }.must_raise ArgumentError
+    end
+    
+    it "raises ArgumentError if user creates a block with more than 5 rooms" do
+      rooms = [2, 3, 4, 5, 7, 10]
+      expect {
+        @hotel_system.create_hotel_block(rooms: rooms, date_range: @date_range, discount: @discount_rate)
+      }.must_raise ArgumentError
+      
+    end
+    
+    it "raises ArgumentError if user creates a block that has duplicate room" do
+      rooms = [2, 3, 4, 4, 7]
+      expect {
+        @hotel_system.create_hotel_block(rooms: rooms, date_range: @date_range, discount: @discount_rate)
+      }.must_raise ArgumentError
+    end
+    
+    it "raises ArgumentError if user creates a block that has nonexistent room" do
+      rooms = [2, 3, 4, 4, 27]
+      expect {
+        @hotel_system.create_hotel_block(rooms: rooms, date_range: @date_range, discount: @discount_rate)
+      }.must_raise ArgumentError
+    end
+  end
+  
+  describe "available_rooms_in_block method" do
+    before do 
+      @rooms = [2, 3, 4]
+      @hotel_system.create_hotel_block(rooms: @rooms, date_range: @date_range)
+      @block_id = @hotel_system.hotel_blocks.first.id
+    end
+    
+    it "returns an array of available room of a given block" do
+      available_rooms = @hotel_system.available_rooms_in_block(@block_id)
+      expect (available_rooms).must_equal @rooms
+    end
+    
+    it "raises ArgumentError if the given block doesn't exist" do
+      block_id = 300  
+      expect {@hotel_system.available_rooms_in_block(block_id)}.must_raise ArgumentError
+    end
+    
+    it "returns empty array if the given block doesn't have any available room" do
+      available_rooms = @hotel_system.available_rooms_in_block(@block_id)
+      expect (available_rooms).must_equal @rooms
+      
+      @rooms.each do |room|
+        @hotel_system.reserve_room(room)
+      end
+      expect (@hotel_system.available_rooms_in_block(@block_id)).must_be_empty
+    end
+  end
+  
+  describe "available_rooms_in_block method" do
```

Comment:
Minor nitpick: This should probably be named something else

Reply:
oh, I missed this line. It should be "reserve_room for rooms in block"